lua/starfall/libs_sh/vectors.lua

```diff
@@ -388,26 +388,43 @@ function vec_methods:normalize()
 	self[3] = self[3] / len
 end
 
+local math_sin, math_cos = math.sin, math.cos
+local deg2rad = math.pi/180
+
 --- Rotate the vector by Angle b.
 -- Self-Modifies. Does not return anything
 -- @param Angle b Angle to rotate by.
 function vec_methods:rotate(b)
-	local vec = unwrap(self)
-	vec:Rotate(aunwrap(b))
+	local vec, ang = unwrap(self), aunwrap(b) 
+
+	local y, p, r = ang.yaw * deg2rad, ang.pitch * deg2rad, ang.roll * deg2rad
+	local ysin, ycos, psin, pcos, rsin, rcos = math_sin(y), math_cos(y), math_sin(p), math_cos(p), math_sin(r), math_cos(r)
+	local psin_rsin, psin_rcos = psin*rsin, psin*rcos
+	local x, y, z = self[1], self[2], self[3]
 
-	self[1] = vec.x
-	self[2] = vec.y
-	self[3] = vec.z
+	self[1] = x * (ycos * pcos) + y * (ycos * psin_rsin - ysin * rcos) + z * (ycos * psin_rcos + ysin * rsin)
+	self[2] = x * (ysin * pcos) + y * (ysin * psin_rsin + ycos * rcos) + z * (ysin * psin_rcos - ycos * rsin)
+	self[3] = x * (-psin) + y * (pcos * rsin) + z * (pcos * rcos)
 end
 
+
+local vec_rotated = {}
```

Comment:
I know you're aiming for performance, but this is no good:
1. You're trying to wrap it multiple times, that'll error.
2. It'll only allow you to have one rotated vector.

Just return a new wrapped table.